components/layouts/mainLayout.tsx

```diff
@@ -30,6 +32,22 @@ export const MainLayout: FC<PropsWithChildren<MainLayoutProps>> = ({
     userType === 'DOCTOR' ? doctorNavigation : patientNavigation
 
   const userName = 'Tom Cook'
+
+  const { mutate } = useMutation({
+    mutationFn: logoutUser,
+    onSuccess: () => {
+      localStorage.removeItem('token')
+      router.push('/login')
+    },
+    onError: (error) => {
+      console.log('error' ,error)
+    }
+  })
+
+  const onLogoutClick = (e: React.MouseEvent<HTMLButtonElement>) => {
```

Comment:
if you do these handlers inline you don't need to type the event ` React.MouseEvent<HTMLButtonElement>` and just let typescript infer the type for you